bftengine/tests/bcstatetransfer/bcstatetransfer_tests.cpp

```diff
@@ -1984,6 +1981,39 @@ TEST_F(BcStTest, dstEnterInternalCycleOnFetchReservedPagesNotReplied) {
                                    testState_.maxRequiredBlockId));
 }
 
+// This test makes sure that current primary is not modified during GettingCheckpointSummaries
+// Its doing so by checking the value of currentPrimary in source selector.
+// It sends 2 cycles of preprepare messages that in any other St state whould have triggerred a new primary awarness.
+// Since they are sent during GettingCheckpointSummaries, they are being  ignored.
+TEST_F(BcStTest, dstTestprimaryAwarnessduringAskForCheckpointSummariesMsg) {
+  ASSERT_NFF(initialize());
+  ASSERT_NFF(dstStartRunningAndCollecting());
+  auto ss = stDelegator_->getSourceSelector();
+  ASSERT_EQ(ss.currentPrimary(), NO_REPLICA);  // make sure primary unknown
+  // ST cycle has started, askForCheckpointSummariesMsg sent and destination is waiting for reply.
+  // Before replying - Generate prePrepare messages to trigger source selector to change the source to avoid primary.
+  ASSERT_EQ(ss.currentPrimary(), NO_REPLICA);  // make sure primary unknown
```

Comment:
Nit: This duplicates the line above.

Reply:
You are right, I forgot to move the line after refactored a new function dstReportMultiplePrePrepareMessagesReceived.

Thanks for letting me know.